samples: drivers: counter: alarm: sample.drivers.counter.alarm

**Describe the bug**
The samples/drivers/counter/alarm/sample.drivers.counter.alarm fails to compile

Observed for:
- nrf52840dk_nrf52840
- nrf9160dk_nrf9160
- nrf52dk_nrf52832

**To Reproduce**
Steps to reproduce the behavior:
1. have nrf52840dk_nrf52840 connected
2. go to your zephyr dir
3. call ./scripts/twister -T samples/drivers/counter/alarm -p nrf52840dk_nrf52840 --device-testing --device-serial /dev/ttyACM0 -v --inline-logs
4. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
```
[2023-03-21T19:34:48.697Z] [0mIn file included from ../../../../../../../../../zephyr/zephyr/include/zephyr/toolchain/gcc.h:92,[0m
[2023-03-21T19:34:48.697Z] [0mfrom ../../../../../../../../../zephyr/zephyr/include/zephyr/toolchain.h:50,[0m
[2023-03-21T19:34:48.697Z] [0mfrom ../../../../../../../../../zephyr/zephyr/include/zephyr/kernel_includes.h:19,[0m
[2023-03-21T19:34:48.697Z] [0mfrom ../../../../../../../../../zephyr/zephyr/include/zephyr/kernel.h:17,[0m
[2023-03-21T19:34:48.697Z] [0mfrom ../../../../../../../../../zephyr/zephyr/samples/drivers/counter/alarm/src/main.c:7:[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/samples/drivers/counter/alarm/src/main.c: In function 'main':[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/include/zephyr/device.h:83:41: error: '__device_dts_ord_89' undeclared (first use in this function); did you mean '__device_dts_ord_19'?[0m
[2023-03-21T19:34:48.697Z] [0m83 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)[0m
[2023-03-21T19:34:48.697Z] [0m|                                         ^~~~~~~~~[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/include/zephyr/toolchain/common.h:132:26: note: in definition of macro '_DO_CONCAT'[0m
[2023-03-21T19:34:48.697Z] [0m132 | #define _DO_CONCAT(x, y) x ## y[0m
[2023-03-21T19:34:48.697Z] [0m|                          ^[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/include/zephyr/device.h:83:33: note: in expansion of macro '_CONCAT'[0m
[2023-03-21T19:34:48.697Z] [0m83 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)[0m
[2023-03-21T19:34:48.697Z] [0m|                                 ^~~~~~~[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/include/zephyr/device.h:209:37: note: in expansion of macro 'DEVICE_NAME_GET'[0m
[2023-03-21T19:34:48.697Z] [0m209 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_ID(node_id))[0m
[2023-03-21T19:34:48.697Z] [0m|                                     ^~~~~~~~~~~~~~~[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/include/zephyr/device.h:226:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'[0m
[2023-03-21T19:34:48.697Z] [0m226 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))[0m
[2023-03-21T19:34:48.697Z] [0m|                                  ^~~~~~~~~~~~~~~~~~[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/samples/drivers/counter/alarm/src/main.c:87:50: note: in expansion of macro 'DEVICE_DT_GET'[0m
[2023-03-21T19:34:48.697Z] [0m87 |         const struct device *const counter_dev = DEVICE_DT_GET(TIMER);[0m
[2023-03-21T19:34:48.697Z] [0m|                                                  ^~~~~~~~~~~~~[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/include/zephyr/device.h:83:41: note: each undeclared identifier is reported only once for each function it appears in[0m
[2023-03-21T19:34:48.697Z] [0m83 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)[0m
[2023-03-21T19:34:48.697Z] [0m|                                         ^~~~~~~~~[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/include/zephyr/toolchain/common.h:132:26: note: in definition of macro '_DO_CONCAT'[0m
[2023-03-21T19:34:48.697Z] [0m132 | #define _DO_CONCAT(x, y) x ## y[0m
[2023-03-21T19:34:48.697Z] [0m|                          ^[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/include/zephyr/device.h:83:33: note: in expansion of macro '_CONCAT'[0m
[2023-03-21T19:34:48.697Z] [0m83 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)[0m
[2023-03-21T19:34:48.697Z] [0m|                                 ^~~~~~~[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/include/zephyr/device.h:209:37: note: in expansion of macro 'DEVICE_NAME_GET'[0m
[2023-03-21T19:34:48.697Z] [0m209 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_ID(node_id))[0m
[2023-03-21T19:34:48.697Z] [0m|                                     ^~~~~~~~~~~~~~~[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/include/zephyr/device.h:226:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'[0m
[2023-03-21T19:34:48.697Z] [0m226 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))[0m
[2023-03-21T19:34:48.697Z] [0m|                                  ^~~~~~~~~~~~~~~~~~[0m
[2023-03-21T19:34:48.697Z] [0m../../../../../../../../../zephyr/zephyr/samples/drivers/counter/alarm/src/main.c:87:50: note: in expansion of macro 'DEVICE_DT_GET'[0m
[2023-03-21T19:34:48.697Z] [0m87 |         const struct device *const counter_dev = DEVICE_DT_GET(TIMER);[0m
[2023-03-21T19:34:48.697Z] [0m|                                                  ^~~~~~~~~~~~~[0m
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.1 LTS
 - Toolchain Zephyr SDK 0.15.2
 - Commit SHA or Version used: zephyr-v3.3.0-1420-g9889e8e20a4f

--- a/drivers/timer/Kconfig.nrf_rtc
+++ b/drivers/timer/Kconfig.nrf_rtc
@@ -8,6 +8,7 @@ config NRF_RTC_TIMER
 	depends on CLOCK_CONTROL
 	depends on SOC_COMPATIBLE_NRF
 	select TICKLESS_CAPABLE
+	depends on !$(dt_nodelabel_enabled,rtc1)
 	help
 	  This module implements a kernel device driver for the nRF Real Time
 	  Counter NRF_RTC1 and provides the standard "system clock driver"

--- a/samples/drivers/counter/alarm/Kconfig
+++ b/samples/drivers/counter/alarm/Kconfig
@@ -3,10 +3,6 @@
 # Copyright (c) 2019 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
-config COUNTER_RTC0
-	bool
-	default y if SOC_FAMILY_NRF
-
 config COUNTER_SAM0_TC32
 	bool
 	default y if BOARD_ATSAMD20_XPRO

--- a/samples/drivers/counter/alarm/boards/nrf51dk_nrf51422.overlay
+++ b/samples/drivers/counter/alarm/boards/nrf51dk_nrf51422.overlay
@@ -1,0 +1,3 @@
+&rtc0 {
+	status = "okay";
+};

--- a/samples/drivers/counter/alarm/boards/nrf52840dk_nrf52840.overlay
+++ b/samples/drivers/counter/alarm/boards/nrf52840dk_nrf52840.overlay
@@ -1,0 +1,3 @@
+&rtc0 {
+	status = "okay";
+};

--- a/samples/drivers/counter/alarm/boards/nrf52dk_nrf52832.overlay
+++ b/samples/drivers/counter/alarm/boards/nrf52dk_nrf52832.overlay
@@ -1,0 +1,3 @@
+&rtc0 {
+	status = "okay";
+};

--- a/samples/drivers/counter/alarm/boards/nrf9160dk_nrf9160.overlay
+++ b/samples/drivers/counter/alarm/boards/nrf9160dk_nrf9160.overlay
@@ -1,0 +1,3 @@
+&rtc0 {
+	status = "okay";
+};

--- a/samples/drivers/counter/alarm/src/main.c
+++ b/samples/drivers/counter/alarm/src/main.c
@@ -21,7 +21,7 @@ struct counter_alarm_cfg alarm_cfg;
 #define TIMER DT_NODELABEL(tc0)
 #elif defined(CONFIG_COUNTER_MICROCHIP_MCP7940N)
 #define TIMER DT_NODELABEL(extrtc0)
-#elif defined(CONFIG_COUNTER_RTC0)
+#elif defined(CONFIG_COUNTER_NRF_RTC)
 #define TIMER DT_NODELABEL(rtc0)
 #elif defined(CONFIG_COUNTER_TIMER_STM32)
 #define TIMER DT_INST(0, st_stm32_counter)

